Source/WebCore/rendering/svg/legacy/LegacyRenderSVGResourceClipper.cpp

```diff
@@ -154,8 +155,17 @@ bool LegacyRenderSVGResourceClipper::pathOnlyClipping(GraphicsContext& context,
     // The SVG specification wants us to clip everything, if clip-path doesn't have a child.
     if (clipPath.isEmpty())
         clipPath.addRect(FloatRect());
+
+    OptionSet<ApplyResult> result = { ApplyResult::ResourceApplied };
+    if (auto* shapeRenderer = dynamicDowncast<LegacyRenderSVGShape>(renderer); shapeRenderer && shapeRenderer->shapeType() == LegacyRenderSVGShape::ShapeType::Rectangle) {
+        // When clipping a rect with a path, if we know the path is entirely inside the rect, we can skip a clip when filling the rect.
```

Comment:
Nice optimization! This logic is now missing for LBSE -- did you consider applying the same optimization for RenderSVGResourceClipper, too?